relation/index.js

```diff
@@ -108,7 +110,25 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+          var done = this.async();
+
+          if (validateName(value) !== true) {
+            done(validateName(value));
```

Comment:
Don't call `validateName` twice - either save the result to a variable, or call `done(false)` here on L118.